### PR TITLE
Add CI for support-ar6.0.z

### PR DESCRIPTION
**Pull request by akariiijima**
### Goodbye TravisCI, Welcome GithubActions
TravisCI is currently not working. This is probably because it was not maintained. Therefore, we will adopt GithubActions instead of using TravisCI.

### CI for support-ar6.0.z
Matrix build test for ruby 2.4-2.7 ✖️ rails 5.0.0-6.0.6

### Fix gemfiles for some reasons about dependencies
- Add gem rails for `require "rails"` in spec/spec_helper.rb
- Use gem mysql2 0.4.x less than rails 5.1.6, and use gem mysql2 0.5.x more than rails 5.1.6
  - https://github.com/rails/rails/blob/5-1-stable/activerecord/CHANGELOG.md#rails-516-march-29-2018
- In ruby 2.4 or less, there are several gem dependencies
  - nokogiri
    - `bundle install` will install a native gem without any additional action in more then ruby 2.5, but in ruby 2.4 or less, need to add gem nokogiri.
      - https://github.com/sparklemotion/nokogiri
    - With the addition of the above adding gem nokogiri 1.10, loofah should be pin version 2.20.2
      - https://github.com/flavorjones/loofah/issues/266
      - https://github.com/getsentry/sentry-ruby/pull/2041
  - dalli
    - Add gem dalli for `require "dalli"` in spec/active_record/turntable/sequencer/katsubushi_spec.rb 
    - dalli should be pin version 2
    - https://github.com/petergoldstein/dalli/issues/758

### Some jobs failed because... 

- More than rails 6 support only ruby more than 2.5, so ruby 2.4 or less is failed CI

### CODE REVIEW

File: gemfiles/rails5_0_0.gemfile [hunk 1]
```diff
@@ -1,10 +1,14 @@
 source "https://rubygems.org"
 
+gem "rails", "5.0.0"
```
Comment (hirocaster) [inline gemfiles/rails5_0_0.gemfile:3]: Why did you need to add a gem such as Rails?

File: gemfiles/rails5_1_4.gemfile [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+gem "railties", "5.1.4"
 gem "activerecord", "5.1.4"
 gem "activesupport", "5.1.4"
 gem "actionview", "5.1.4"
-
 gem "mysql2", "~> 0.4.4"
 
-gem "minitest", "< 5.3.4"
-gem "mocha", "~> 0.14", require: false
+gem "nokogiri", "~> 1.10" if RUBY_VERSION.to_f < 2.5
```
Comment (hirocaster) [inline gemfiles/rails5_1_4.gemfile:10]: Why are these making changes?

**Comment by hirocaster** [comment]
Glad you sent us a PR, and we will be sure to keep you posted on our progress. 😄 
However, it would be very helpful if you could explain the reasons for the increase in dependence as it seems to be increasing.

**Comment by akariiijima** [comment]
@hirocaster 
I added descriptions about fix reasons! 
Please check and let me know if you have some questions.

**Comment by hirocaster** [comment]
https://github.com/mixigroup/activerecord-turntable/pull/18#discussion_r1239323534

> Why did you need to add a gem such as Rails?

I looked at the DESCRIPTION but could not find a reason for this.

**Comment by akariiijima** [comment]
About adding "gem rails", I referenced original file (ex: gemfiles/rails6_0_4.gemfile, etc...)😄

**Comment by hirocaster** [comment]
Okay, I understand.
I recognized that I may need to organize gemfile as it should not inherently depend on Rails.

**Comment by akariiijima** [comment]
`gem rails` is needed for `require rails`
https://github.com/mixigroup/activerecord-turntable/blob/3d4a955b9ef7e15765a0ef0ec4670413939a1573/spec/spec_helper.rb#L6
Gemfiles seem to be need not to change in this last commit. :+1:

File: gemfiles/rails5_0_0.gemfile (hunk 1 above)
Comment (hirocaster) [reply gemfiles/rails5_0_0.gemfile:3]: I asked to be listed in the DESCRIPTION.

File: gemfiles/rails5_1_4.gemfile (hunk 2 above)
Comment (hirocaster) [reply gemfiles/rails5_1_4.gemfile:10]: I asked to be listed in the DESCRIPTION.

Comment (hirocaster) [reply gemfiles/rails5_1_4.gemfile:10]: https://github.com/mixigroup/activerecord-turntable/pull/18#issuecomment-1609045294

File: gemfiles/rails5_0_0.gemfile (hunk 1 above)
Comment (akariiijima) [reply gemfiles/rails5_0_0.gemfile:3]: https://github.com/mixigroup/activerecord-turntable/pull/18#issuecomment-1610825952

File: spec/active_record/turntable/active_record_ext/query_cache_spec.rb [hunk 3]
```diff
@@ -6,16 +6,18 @@ def middleware(&app)
...
-        original_handlers = ActiveRecord::Base.connection_handlers
-        ActiveRecord::Base.connection_handlers = { writing: ActiveRecord::Base.default_connection_handler, reading: ActiveRecord::ConnectionAdapters::ConnectionHandler.new }
-      end
-      executor.wrap do
-        app.call(env)
-      end
-    ensure
-      if ActiveRecord::Turntable::Util.ar60_or_later?
-        ActiveRecord::Base.connection_handlers = original_handlers
+      begin
```
Comment (hirocaster) [inline spec/active_record/turntable/active_record_ext/query_cache_spec.rb]: This commit is duplicate #19.
Please, fix the change commits.

**Review by hirocaster — CHANGES_REQUESTED** [review]

File: spec/active_record/turntable/active_record_ext/query_cache_spec.rb (hunk 3 above)
Comment (akariiijima) [reply spec/active_record/turntable/active_record_ext/query_cache_spec.rb]: I fixed!
